File: sdks/python/src/bigrag/resources/admin/access.py
from __future__ import annotations

from typing import TYPE_CHECKING

from bigrag.resources.admin._shared import _pagination
from bigrag.types.access import AccessLogListResponse, AccessLogOverviewResponse

if TYPE_CHECKING:
    from bigrag._core import BigRAGCore


class AdminAccessResource:

    def __init__(self, client: BigRAGCore) -> None:
        self._client = client

    async def logs(
        self,
        *,
        action: str | None = None,
        actor_id: str | None = None,
        collection: str | None = None,
        method: str | None = None,
        path: str | None = None,
        status_family: str | None = None,
        success: bool | None = None,
        limit: int | None = None,
        offset: int | None = None,
    ) -> AccessLogListResponse:
        params = _pagination(limit=limit, offset=offset)
        if action is not None:
            params["action"] = action
        if actor_id is not None:
            params["actor_id"] = actor_id
        if collection is not None:
            params["collection"] = collection
        if method is not None:
            params["method"] = method
        if path is not None:
            params["path"] = path
        if status_family is not None:
            params["status_family"] = status_family
        if success is not None:
            params["success"] = "true" if success else "false"
        return await self._client._request(
            "GET", "/v1/admin/access/logs", params=params
        )

    async def overview(
        self, *, window_days: int | None = None
    ) -> AccessLogOverviewResponse:
        params: dict[str, str] = {}
        if window_days is not None:
            params["window_days"] = str(window_days)
        return await self._client._request(
            "GET", "/v1/admin/access/overview", params=params
        )

File: sdks/typescript/src/resources/admin/access.ts
import type { RequestClient } from "../../core.js";
import type { AccessLogListResponse, AccessLogOverviewResponse } from "../../types/index.js";
import { pagination } from "./_shared.js";

export class AdminAccessResource {
  constructor(private readonly _client: RequestClient) {}

  logs(
    options: {
      action?: string;
      actorId?: string;
      collection?: string;
      method?: string;
      path?: string;
      statusFamily?: string;
      success?: boolean;
      limit?: number;
      offset?: number;
    } = {},
  ): Promise<AccessLogListResponse> {
    const params = pagination(options);
    if (options.action !== undefined) params.action = options.action;
    if (options.actorId !== undefined) params.actor_id = options.actorId;
    if (options.collection !== undefined) params.collection = options.collection;
    if (options.method !== undefined) params.method = options.method;
    if (options.path !== undefined) params.path = options.path;
    if (options.statusFamily !== undefined) params.status_family = options.statusFamily;
    if (options.success !== undefined) params.success = String(options.success);
    return this._client._request("GET", "/v1/admin/access/logs", { params });
  }

  overview(options: { windowDays?: number } = {}): Promise<AccessLogOverviewResponse> {
    const params: Record<string, string> = {};
    if (options.windowDays !== undefined) params.window_days = String(options.windowDays);
    return this._client._request("GET", "/v1/admin/access/overview", { params });
  }
}
